src/pytorch_lightning/lite/lite.py

```diff
@@ -65,376 +84,238 @@ class LightningLite(ABC):
             or bfloat16 precision (``"bf16"``).
         plugins: One or several custom plugins
         gpus: Provides the same function as the ``devices`` argument but implies ``accelerator="gpu"``.
+
+            .. deprecated:: v1.8.0
+                ``gpus`` has been deprecated in v1.8.0 and will be removed in v1.10.0.
+                Please use ``accelerator='gpu'`` and ``devices=x`` instead.
+
         tpu_cores: Provides the same function as the ``devices`` argument but implies ``accelerator="tpu"``.
+
+            .. deprecated:: v1.8.0
+                ``tpu_cores`` has been deprecated in v1.8.0 and will be removed in v1.10.0.
+                Please use ``accelerator='tpu'`` and ``devices=x`` instead.
     """
 
     def __init__(
         self,
-        accelerator: Optional[Union[str, Accelerator]] = None,
-        strategy: Optional[Union[str, Strategy]] = None,
+        accelerator: Optional[Union[str, PLAccelerator]] = None,
+        strategy: Optional[Union[str, PLStrategy]] = None,
         devices: Optional[Union[List[int], str, int]] = None,
         num_nodes: int = 1,
         precision: Union[int, str] = 32,
-        plugins: Optional[Union[_PLUGIN_INPUT, List[_PLUGIN_INPUT]]] = None,
+        plugins: Optional[Union[_PL_PLUGIN_INPUT, List[_PL_PLUGIN_INPUT]]] = None,
         gpus: Optional[Union[List[int], str, int]] = None,
         tpu_cores: Optional[Union[List[int], str, int]] = None,
     ) -> None:
-        self._connector = _Connector(
+
+        if gpus is not None or tpu_cores is not None:
+            devices, accelerator = _convert_deprecated_device_flags(
+                accelerator=accelerator,
+                devices=devices,
+                gpus=gpus,
+                tpu_cores=tpu_cores,
+            )
+
+        lite_plugins: Optional[Union[_LITE_PLUGIN_INPUT, List[_LITE_PLUGIN_INPUT]]]
+        if isinstance(plugins, PLPrecisionPlugin):
+            lite_plugins = _to_lite_precision_plugin(plugins)
+        elif isinstance(plugins, list):
+            lite_plugins = [
+                _to_lite_precision_plugin(plugin) if isinstance(plugin, PLPrecisionPlugin) else plugin  # type: ignore[misc]  # noqa: E501
```

Comment:
Mypy has a problem here with the list comprehension that I couldn't resolve. Maybe @carmocca has an idea.

Reply:
`_to_lite_precision_plugin` supports returning an Optional, and lite_plugins is meant to be a list of plugins, not a list of optional plugins.

There's multiple ways you can fix it. I would not allow optional returns from `_to_lite_precision_plugin`

Reply:
Thanks, yeah that makes sense. addressed